integration/spark/shared/src/main/java/io/openlineage/spark/agent/lifecycle/plan/KustoRelationVisitor.java

```diff
@@ -0,0 +1,161 @@
+package io.openlineage.spark.agent.lifecycle.plan;
+
+import io.openlineage.client.OpenLineage;
+import io.openlineage.spark.api.DatasetFactory;
+import io.openlineage.spark.api.OpenLineageContext;
+import io.openlineage.spark.api.QueryPlanVisitor;
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+import lombok.extern.slf4j.Slf4j;
+import org.apache.commons.lang3.reflect.FieldUtils;
+import org.apache.spark.api.java.Optional;
+import org.apache.spark.sql.catalyst.plans.logical.LogicalPlan;
+import org.apache.spark.sql.execution.datasources.LogicalRelation;
+import org.apache.spark.sql.sources.BaseRelation;
+import org.apache.spark.sql.sources.CreatableRelationProvider;
+import org.apache.spark.sql.types.StructType;
+
+/** */
+@Slf4j
+public class KustoRelationVisitor<D extends OpenLineage.Dataset>
+    extends QueryPlanVisitor<LogicalRelation, D> {
+
+  private final DatasetFactory<D> factory;
+  private static final String KUSTO_CLASS_NAME =
+      "com.microsoft.kusto.spark.datasource.KustoRelation";
+
+  private static final String KUSTO_PROVIDER_CLASS_NAME =
+      "com.microsoft.kusto.spark.datasource.DefaultSource"; // Not used
+
+  private static final String KUSTO_URL_SUFFIX = ".kusto.windows.net";
+
+  private static final String KUSTO_PREFIX = "azurekusto://";
+
+  public KustoRelationVisitor(OpenLineageContext context, DatasetFactory<D> factory) {
+    super(context);
+    this.factory = factory;
+  }
+
+  public static boolean isKustoClass(LogicalPlan plan) {
+    return plan instanceof LogicalRelation
+        && ((LogicalRelation) plan).relation().getClass().getName().equals(KUSTO_CLASS_NAME);
+  }
+
+  public static boolean isKustoSource(CreatableRelationProvider provider) {
+    return provider.getClass().getName().equals(KUSTO_PROVIDER_CLASS_NAME);
+  }
+
+  public static boolean hasKustoClasses() {
+    try {
+      KustoRelationVisitor.class
+          .getClassLoader()
+          .loadClass("com.microsoft.kusto.spark.datasource.DefaultSource");
+      return true;
+    } catch (Exception e) {
+      // swallow - we don't care
+    }
+    try {
+      Thread.currentThread()
+          .getContextClassLoader()
+          .loadClass("com.microsoft.kusto.spark.datasource.DefaultSource");
+      return true;
+    } catch (Exception e) {
+      // swallow - we don't care
+    }
+
+    return false;
```

Comment:
We should generalize this so we don't have to keep copy/pasting the same boilerplate.

Reply:
That's a great idea. Do you suggest we do that as part of this PR or a separate one @collado-mike? Seeing as it would affect the other datasources' relationVisitor implementations too.

Reply:
@collado-mike - This one is marked as changes requested.  Do you still think it's necessary for this PR to amend other data sources or can we mark this as a separate issue and Hanna and I can look into doing a larger cleanup?

Reply:
I'll mark this statement resolved